### DISCUSSION TITLE
AWS: adding a custom domain name to user pool domain and setting route53 record in parralel.

### DISCUSSION THREAD

**Discussion by Jimmy89** [pulumi/pulumi#7506] [Pulumi Providers] [answered]
I try to create a user pool domain with a custom domain name. However, setting the route53 record while creating the user pool domain is not possible as both wait for each other.

## Steps to reproduce
I want to create a user pool domain with a custom domain, using our valid certificate within certificate manager. The problem is that pulumi awaits on the user pool domain to be finished with creation, before creating the route53 record. However, AWS in it's turn is waiting for the route53 record to exists. In order to create a route53 record I need the `cloudfrontDistributionArn` output from the user pool domain creation. Therefore, they wait for each other until I manually intervene.

I did not find any pulumi documentation about setting these things up. Based [on this topic](https://stackoverflow.com/a/64796658) I created the following code:

```
const poolDomain = new aws.cognito.UserPoolDomain("domain", {
  domain: `DOMAINNAME`,
  userPoolId: pool.id,
  certificateArn: "arn:aws:acm:us-east-1:551235321111:certificate/XXXXXXXXXXXXXXX",
});

new aws.route53.Record("userPoolDomainDnsRecord", {
  name: `DOMAINNAME`,
  type: "A",
  zoneId: dnsZone.then((zone) => zone.zoneId),
  aliases: [{
    evaluateTargetHealth: false,
    name: poolDomain.cloudfrontDistributionArn,
    zoneId: "Z2FDTNDATAQYW2",
  }],
});
``` 

Is there a way to circumvent this issue?

Expected: Route53 record is created without waiting for complete initialization of user pool domain.

Actual: Route53 waits for user pool domain to be created.

**Comment by lukehoban** [comment, marked as answer]
You should be able to apply the same workaround you linked to in https://stackoverflow.com/questions/51249583/cognito-own-domain-name-required-a-record/64796658#64796658 to this in Pulumi.  Have you tried the approach suggested there?  As noted in that reply, you will need to break the cycle between these - as fundamentally neither resource in AWS can succesfully create without the other as written above.  That post includes an approach that does break the cycle, apparently by introducing a subdomain for the Cognito UserPool.
